packages/nuxt/src/app/nuxt.ts

```diff
@@ -492,14 +492,18 @@ export function isNuxtPlugin (plugin: unknown) {
  * @since 3.0.0
  */
 export function callWithNuxt<T extends (...args: any[]) => any> (nuxt: NuxtApp | _NuxtApp, setup: T, args?: Parameters<T>) {
+  return nuxt.vueApp.runWithContext(() => callWithNuxtContext(nuxt, setup, args))
+}
+
+export function callWithNuxtContext<T extends (...args: any[]) => any> (nuxt: NuxtApp | _NuxtApp, setup: T, args?: Parameters<T>) {
```

Comment:
we probably need to add `callWithNuxtContext` to `unctx` transforms list

Reply:
May I ask what the `unctx` transforms list refers to?

Reply:
you can take a look here https://github.com/nuxt/nuxt/blob/160efe7a7fe0a65755d9f7181a2a7f5b2532b66f/packages/nuxt/src/core/nuxt.ts#L396

It's needed server side to avoid cross request state pollution for node < 20 (or 22?) and client side to correctly keep context in promises as async context isn't supported yet in JS

Reply:
We probably don't need to, callWithNuxt uses `vueApp.runWithContext`